vms/platformvm/vm.go

```diff
@@ -245,6 +245,15 @@ func (vm *VM) Initialize(
 	// [periodicallyPruneMempool] grabs the context lock.
 	go vm.periodicallyPruneMempool(execConfig.MempoolPruneFrequency)
 
+	go func() {
+		err := vm.state.ReindexBlocks(&vm.ctx.Lock, vm.ctx.Log)
+		if err != nil {
+			vm.ctx.Log.Warn("reindexing blocks failed",
```

Comment:
This is marked as a `Warn` because shutting down the node mid-way through the indexing will report an error here. There isn't an easy way to exit this routine during `Shutdown`, because `Shutdown` has the context lock held, and this goroutine may be blocked on grabbing that lock.